### PR TITLE
Dashboard: localize chart axis and chart markers

### LINKED ISSUE #621: Stats: dates at the bottom of the chart are not localized
**Describe the bug**
On "My Store" screen, the dates displayed at the bottom of the stats are not localized.
image

**Mobile Environment**
Please include:
- Device: iPhone 8
- iOS version: 12.1.2
- WooCommerce iOS version: 1.0.0.1

### PR DESCRIPTION
**Pull request by bummytime**
This PR fixes #621 by localizing the chart axis labels:

3g
3j
3r

As well as the chart markers on the chart bars:

## Testing

* Build and run the app — verify the unit tests are ✅ and the code deltas look 👍 
* Build and run the app using different device languages in the simulator. You can do this by changing the schema settings in Xcode:

xcode_schema_settings

Verify the chart markers and axis labels look like they are localized.

@mindgraffiti would you mind taking a peek at this?

----
Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift [hunk 1]
```diff
@@ -8,43 +8,74 @@ extension DateFormatter {
...
 
         /// Date formatter used for creating the date displayed on a chart axis for **week** granularity.
         ///
-        public static let chartsWeekFormatter: DateFormatter = {
+        public static let chartAxisWeekFormatter: DateFormatter = {
             let formatter = DateFormatter()
-            formatter.locale = Locale(identifier: "en_US_POSIX")
-            formatter.timeZone = TimeZone(identifier: "GMT")
-            formatter.dateFormat = "MMM d"
+            formatter.setLocalizedDateFormatFromTemplate("MMM d")
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift]: Can you add a comment above this, reminding me about what this format looks like?

// e.g., Jan 23

File: WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift [hunk 2]
```diff
@@ -8,43 +8,74 @@ extension DateFormatter {
...
 
         /// Date formatter used for creating the date displayed on a chart axis for **month** granularity.
         ///
-        public static let chartsMonthFormatter: DateFormatter = {
+        public static let chartAxisMonthFormatter: DateFormatter = {
             let formatter = DateFormatter()
-            formatter.locale = Locale(identifier: "en_US_POSIX")
-            formatter.timeZone = TimeZone(identifier: "GMT")
-            formatter.dateFormat = "MMM"
+            formatter.setLocalizedDateFormatFromTemplate("MMM")
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift]: // e.g., Jan

Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift]: same comment request

File: WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift [hunk 3]
```diff
@@ -8,43 +8,74 @@ extension DateFormatter {
...
+            formatter.setLocalizedDateFormatFromTemplate("MMM d")
+            return formatter
+        }()
+
+        /// Date formatter used for creating the date displayed on a chart marker for **week** granularity.
+        ///
+        public static let chartMarkerWeekFormatter: DateFormatter = {
+            let formatter = DateFormatter()
+            // Note: Passing "w Y" into `setLocalizedDateFormatFromTemplate()` will result in an empty string, however
+            // simply passing in "w" works. So that is what we have to unfortunately do here.
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/DateFormatter+Helpers.swift]: excellent documentation +1

File: WooCommerce/WooCommerce.xcodeproj/xcshareddata/xcschemes/WooCommerce.xcscheme [hunk 4]
```diff
@@ -85,6 +85,7 @@
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      language = "ja"
```
Comment (mindgraffiti) [inline WooCommerce/WooCommerce.xcodeproj/xcshareddata/xcschemes/WooCommerce.xcscheme]: Probably shouldn't commit this

Comment (bummytime) [reply WooCommerce/WooCommerce.xcodeproj/xcshareddata/xcschemes/WooCommerce.xcscheme]: Gah! Agree!

**Pull request merged**